Undo notification close (history) #574

ArijanJ opened the pull request:
Opening this PR early so we can discuss how this would ideally work. It's bound to ctrl+z for now, and so far this method looks promising, although I haven't tested it much.

For groups, I assume we would want it to undo the deletion of the entire group, and I was also curious if it's better to somehow stick the restored notifications to the top of the list. I tried this with the two commented-out lines, but might need to reposition the widget itself if we want to go this route. What do we think?

ArijanJ commented:
Maybe something like this? The problem I'm seeing with this is if you delete 3/10 notifications in a group, ctrl+z will restore all three at once instead of one by one. We might need to somehow check every single deletion and mark it as either a group-delete or a single-delete, which is likely a bit more work

ErikReider reviewed (commented):
Thanks for picking this up! :D

ArijanJ commented on src/controlCenter/controlCenter.vala:
```diff
@@ -651,6 +673,11 @@ namespace SwayNotificationCenter {
+                    notification_history.push_head (HistoryEntry () {
+                        noti = noti,
+                        group_id = group.name_id
+                    });
```
Fixed with 308608f

ArijanJ commented:
I think this simple implementation should be enough for most use cases. I tried implementing full-group restore but it seems to be more complex and I'm not 100% sure on how the events propagate up from each widget/button :(

Also moved it to the `on_released...` callback since holding down the action had some glitches, and added the shortcut to the manpage and readme. Let me know what you think

ArijanJ commented:
Yeah of course, whatever you decide. Do you plan on making UI for browsing through it?
